junit-jupiter-params/src/main/java/org/junit/jupiter/params/aggregator/DefaultArgumentsAccessor.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.params.aggregator;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.params.converter.DefaultArgumentConverter;
+import org.junit.platform.commons.util.Preconditions;
+
+public class DefaultArgumentsAccessor implements ArgumentsAccessor {
+
+	private final Object[] arguments;
+
+	public DefaultArgumentsAccessor(Object[] arguments) {
+		Preconditions.notNull(arguments, "DefaultArgumentsAccessor initialized with null arguments");
+		this.arguments = arguments;
+	}
+
+	public Object get(int index) {
+		Preconditions.condition(index >= 0 && index < arguments.length,
+			String.format("Index must be between 0 and %d", arguments.length));
+		return arguments[index];
+	}
+
+	public Object[] toArray() {
+		return arguments;
+	}
+
+	public List<Object> toList() {
+		return Arrays.asList(arguments);
+	}
+
+	public Character getChar(int index) {
+		return (Character) get(Character.class, index);
+	}
+
+	public Boolean getBoolean(int index) {
+		return (Boolean) get(Boolean.class, index);
+	}
+
+	public Byte getByte(int index) {
+		return (Byte) get(Byte.class, index);
+	}
+
+	public Short getShort(int index) {
+		return (Short) get(Short.class, index);
+	}
+
+	public Integer getInteger(int index) {
+		return (Integer) get(Integer.class, index);
+	}
+
+	public Long getLong(int index) {
+		return (Long) get(Long.class, index);
+	}
+
+	public Float getFloat(int index) {
+		return (Float) get(Float.class, index);
+	}
+
+	public Double getDouble(int index) {
+		return (Double) get(Double.class, index);
+	}
+
+	public String getString(int index) {
+		return (String) get(String.class, index);
+	}
+
+	public Object get(Class<?> clazz, int index) {
```

Comment:
Please make this method generic:

```java
public <T> T get(Class<T> clazz, int index) {
    ...
    return clazz.cast(DefaultArgumentConverter.INSTANCE.convert(arguments[index], clazz));
}
```

This way you don't have cast in any of the more specific `get*`  methods and neither will code using the method.

Reply:
Thanks, looks great!

I am currently doubtful about how to proceed on the following 2 issues

1. Since the ArgumentAccessor is now an interface, instead of providing the DefaultArgumentAccessor i believe there is a need for a way to provide another implementation? In case of DI its straightforwar, just declare the implementation as type. 
In case of `@AggregateWith` , it must be passed somehow to the ArgumentsAggregator

2. Is there a need for an ArgumentAggregationException thrown by ArgumentAggregators and Accessors?

Reply:
> Since the ArgumentAccessor is now an interface, instead of providing the DefaultArgumentAccessor i believe there is a need for a way to provide another implementation?

I don't think that's necessary. The interface in this case is only intended to make it harder to depend on implementation details and give us more clarity and possibly flexibility to change things in the future. In the end, `DefaultArgumentsAccessor` should be annotated with `@API(Internal)`.

> Is there a need for an ArgumentAggregationException thrown by ArgumentAggregators and Accessors?

I was also wondering about that. Let's do it for the sake of consistency with `ArgumentConverter` but I think it should only be used by `ArgumentsAggregator` because an `ArgumentsAccessor` does not aggregate.